DSharpPlus.Commands/CommandsExtension.cs

```diff
@@ -159,6 +172,86 @@ public async ValueTask AddProcessorsAsync(IEnumerable<ICommandProcessor> process
 
     public TProcessor GetProcessor<TProcessor>() where TProcessor : ICommandProcessor => (TProcessor)this._processors[typeof(TProcessor)];
 
+    /// <summary>
+    /// Adds all public checks from the provided assembly to the extension.
+    /// </summary>
+    public void AddChecks(Assembly assembly)
+    {
+        foreach
+        (
+            Type t in assembly.ExportedTypes.Where
+            (
+                candidate => candidate.GetInterfaces().Any
+                (
+                    x => x.FullName == "DSharpPlus.Commands.ContextChecks.IContextCheck`1"
+                )
+            )
+        )
+        {
+            this.AddCheck(t);
+        }
+    }
+
+    /// <summary>
+    /// Adds a new check to the extension.
+    /// </summary>
+    public void AddCheck<T>()
+        where T : IContextCheck
+        => this.AddCheck(typeof(T));
+
+    /// <summary>
+    /// Adds a new check to the extension.
+    /// </summary>
+    public void AddCheck(Type checkType)
+    {
+        // get all implemented check interfaces, we can pretty easily handle having multiple checks in one type
+        IEnumerable<Type> interfaces = checkType.GetInterfaces()
+            .Where(x => x.FullName == "DSharpPlus.Commands.ContextChecks.IContextCheck`1");
+
+        foreach(Type t in interfaces)
+        {
+            Type attributeType = t.GetGenericArguments()[0];
+
+            MethodInfo method = t.GetMethods(BindingFlags.Public | BindingFlags.Static)
+                .Where(x => x.Name == nameof(IContextCheck<ContextCheckAttribute>.ExecuteCheckAsync))
+                .Where(x => x.GetParameters()[0].ParameterType == attributeType)
+                .First();
```

Comment:
Could be done as a single `First` call:
```suggestion
            MethodInfo method = t.GetMethods(BindingFlags.Public | BindingFlags.Static)
                                 .First(x => x.Name == nameof(IContextCheck<ContextCheckAttribute>.ExecuteCheckAsync) && x.GetParameters()[0].ParameterType == attributeType);
```

Reply:
no, that is logically not equivalent. i want to know whether the first parameter follows the criteria, not whether there exists one such parameter

Reply:
on a second note, github provided an excellent case against your use of over-length lines, and this would be *fine*, albeit not with this formatting